stdlib/Random/src/RNGs.jl

```diff
@@ -216,6 +216,216 @@ function reset_caches!(r::MersenneTwister)
     r
 end
 
+
+## Xoshiro RNG
+# Lots of implementation is shared with TaskLocal
+
+"""
+    Xoshiro
+
+Xoshiro256++ is a fast pseudorandom number generator originally developed by Sebastian Vigna.
+Reference implementation is available on on http://prng.di.unimi.it
+
+Apart from the high speed, Xoshiro has a small memory footprint, making it suitable for
+applications where many different random states need to be held for long time.
+
+Julia's Xoshiro implementation has a bulk-generation mode; this seeds new virtual PRNGs
+from the parent, and uses SIMD to generate in parallel (i.e. the bulk stream consists of
+multiple interleaved xoshiro instances).
+The virtual PRNGs are discarded once the bulk request has been serviced (and should cause
+no heap allocations).
+"""
+mutable struct Xoshiro <: AbstractRNG
+    s0::UInt64
+    s1::UInt64
+    s2::UInt64
+    s3::UInt64
+end
+
+Xoshiro(::Nothing) = Xoshiro()
```

Comment:
It will end up using RandomDevice, but I also wonder why this exists. It's only here to copy other RNGs that also have such methods. Similarly, why do we have a `seed!(rng, nothing)` method?

Reply:
`Xoshiro()` randomizes all the fields with `RandomDevice()`; having `nothing` equivalent to no explicitly passed seed can be useful to forward seeds, like in e.g. `myalgo(...; seed=nothing)`. I don't think this is documented though...